src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Statistics/TableBuilderControllerTests.cs

```diff
@@ -221,126 +292,94 @@ public async Task QueryForDataBlock_NoGeoJson()
 
             var result = await controller.QueryForDataBlock(_releaseId, _dataBlockId);
+            VerifyAllMocks(mocks);
 
-            Assert.IsType<NotFoundResult>(result.Result);
+            result.AssertNotFoundResult();
         }
 
         [Fact]
         public async Task QueryForDataBlock_NotDataBlockType()
         {
-            var contentPersistenceHelper =
-                MockPersistenceHelper<ContentDbContext, ReleaseContentBlock>(
-                    new ReleaseContentBlock
+            var releaseContentBlock = new ReleaseContentBlock
+            {
+                ReleaseId = _releaseId,
+                Release = new Release
+                {
+                    Id = _releaseId,
+                    Publication = new Publication
                     {
-                        ReleaseId = _releaseId,
-                        Release = new Release
-                        {
-                            Id = _releaseId,
-                        },
-                        ContentBlockId = _dataBlockId,
-                        ContentBlock = new HtmlBlock
-                        {
-                            Id = _dataBlockId,
-                        }
+                        Id = Guid.NewGuid()
                     }
-                );
+                },
+                ContentBlockId = _dataBlockId,
+                ContentBlock = new HtmlBlock
+                {
+                    Id = _dataBlockId,
+                }
+            };
+            
+            var (controller, mocks) = BuildControllerAndDependencies();
 
-            var controller = BuildTableBuilderController(
-                contentPersistenceHelper: contentPersistenceHelper.Object
-            );
+            SetupCall(mocks.persistenceHelper, releaseContentBlock);
 
-            controller.ControllerContext = new ControllerContext
-            {
-                HttpContext = new DefaultHttpContext()
-            };
+            var exception = await Assert.ThrowsAsync<TargetInvocationException>(() => controller.QueryForDataBlock(_releaseId, _dataBlockId));
+            Assert.IsType<ArgumentException>(exception.InnerException);
+        }
 
-            var result = await controller.QueryForDataBlock(_releaseId, _dataBlockId);
+        [Fact]
+        public void TableBuilderResultViewModel_SerialiseAndDeserialise()
+        {
+            var converted = DeserializeObject<TableBuilderResultViewModel>(SerializeObject(_tableBuilderResults));
+            converted.AssertDeepEquals(_tableBuilderResults);
+        }
 
-            Assert.IsType<NotFoundResult>(result.Result);
+        private static DataBlockTableResultCacheKey IsMatchingDataBlockCacheKey(ReleaseContentBlock releaseContentBlock)
+        {
+            return It.Is<DataBlockTableResultCacheKey>(cacheKey => cacheKey.AssertDeepEquals(new DataBlockTableResultCacheKey(releaseContentBlock)));
```

Comment:
Just wondering if there's a disadvantage to implementing equality method on DataBlockTableResultCacheKey and using `Equals` instead? As there's no complex types equals should be sufficient I think.

Reply:
That sounds good to me, thanks. I was also going to say that with c#9 Record types (when we can use them in the Admin!), I think this particular example would be equal with the default value equality that Record types offer.

Reply:
Nice one - yeah a shame we can't use them in admin yet :cry: